File: src/main/java/com/machfour/macros/objects/FoodType.java
package com.machfour.macros.objects;

import java.util.NoSuchElementException;

public enum FoodType {
    PRIMARY("primary"), COMPOSITE("composite"), USDA("usda"), NUTTAB("nuttab"), SPECIAL("special");

    final String name;

    FoodType(String name) {
        this.name = name;
    }

    public static FoodType fromString(String name) {
        switch (name) {
            case "primary":
                return PRIMARY;
            case "composite":
                return COMPOSITE;
            case "usda":
                return USDA;
            case "nuttab":
                return NUTTAB;
            case "special":
                return SPECIAL;
            default:
                throw new NoSuchElementException("No FoodType with name '" + name + "'.");
        }
    }

    public String getName() {
        return name;
    }

    @Override
    public String toString() {
        return getName();
    }

}

File: src/main/kotlin/com/machfour/macros/objects/FoodType.kt
package com.machfour.macros.objects

enum class FoodType constructor(val niceName: String) {
    PRIMARY("primary"),
    COMPOSITE("composite"),
    USDA("usda"),
    NUTTAB("nuttab"),
    SPECIAL("special");

    override fun toString(): String {
        return niceName
    }

    companion object {
        fun fromString(name: String): FoodType {
            return when (name) {
                "primary" -> PRIMARY
                "composite" -> COMPOSITE
                "usda" -> USDA
                "nuttab" -> NUTTAB
                "special" -> SPECIAL
                else -> throw NoSuchElementException("No FoodType with name '$name'.")
            }
        }
    }

}
